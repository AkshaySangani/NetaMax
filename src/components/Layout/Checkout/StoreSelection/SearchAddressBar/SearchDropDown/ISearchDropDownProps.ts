import { Status } from "use-places-autocomplete";

export interface ISearchDropDownProps {
  /**
   * List of recommended addresses
   * @type {google.maps.places.AutocompletePrediction[]}
   */
  data: google.maps.places.AutocompletePrediction[];

  /**
   * Status
   * @type {Status}
   */
  status: Status;

  /**
   * Handles the select of a searched address
   * @type {(val: string, showVal: string) => void}
   */
  handleSelectAddress: (val: string, showVal: string) => void;
}
