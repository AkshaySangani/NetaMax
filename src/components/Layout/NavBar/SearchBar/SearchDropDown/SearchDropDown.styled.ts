import styled from "styled-components";
import { responsive } from "styled/responsive";

import { Box } from "@chakra-ui/react";

export const StyledDropdown = styled(Box)`
  position: absolute;
  top: ${(props) => (props.$isLargerThanSm ? "0" : "0")};
  left: ${(props) => (props.$isLargerThanSm ? "none" : "0")};
  width: ${(props) => (props.$isLargerThanSm ? "480px" : "100%")};
  height: 50px;
  background-color: #fff;
  margin-top: 55px;
  box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 0, rgba(0, 0, 0, 0.1) 0px 4px 11px;
  border-radius: 5px;
  z-index: 10;
  > ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    > li {
      padding-right: 10px;
      padding-left: 2px;
      justify-content: center;
      align-items: center;
      background-color: #fff;
      border-bottom: 0.5px solid #cbcbcb;
      height: 60px;
      display: flex;
      align-items: center;
      a {
        text-decoration: none;
        -webkit-box-shadow: none;
        -moz-box-shadow: none;
        box-shadow: none;
      }
      &:hover {
        cursor: pointer;
      }
      &:last-child {
        border-bottom-left-radius: 12px;
        border-bottom-right-radius: 12px;
        border-bottom: 0px;
      }
      &:first-child {
        ${responsive.smPlus} {
          border-top-left-radius: 12px;
          border-top-right-radius: 12px;
        }
      }
      &:not(:first-child) {
        border-top: 0;
      }
    }
  }
`;
