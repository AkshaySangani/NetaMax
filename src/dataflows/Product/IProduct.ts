export interface IProduct {
  /**
   * The unique identifier for the item.
   * @type {string}
   */
  id: string;

  /**
   * The name identifies the item.
   * @type {string}
   */
  name: string;

  /**
   * Is a number or code assigned to an item to be able to identify the item.
   * @type {string}
   */
  sku: string;

  /**
   * Are the images to the item.
   * @type {string[]}
   */
  images: string[];

  /**
   * Item creation date in UTC format.
   * @type {Date}
   */
  createdOnUtc: Date;

  /**
   * Item creation date in UTC format.
   * @type {Date}
   */
  updatedOnUtc: Date;

  /**
   * Item image path.
   * @type {string}
   */
  seoFilename: string;

  /**
   * Item price.
   * @type {string}
   */
  price: string;

  /**
   * Item old price.
   * @type {string}
   */
  oldPrice: string;
  /**
   * Item order minimum quantity.
   * @type {number}
   */
  orderMinimumQuantity: number;

  /**
   * Item order maximum quantity.
   * @type {number}
   */
  orderMaximumQuantity: number;

  /**
   * Item pictureId.
   * @type {number}
   */
  pictureId?: number;

  /**
   * Item status.
   * @type {string}
   */
  status?: string;

  /**
   * The product category.
   * @type {string}
   */
  categoryId: string;
  /**
   * The product's stock quantity.
   * @type {number}
   */
  stockQuantity: number;
  /**
   * If it is different than 0 we check on stockQuantity,
   * otherwise, we don't.
   * @type {number}
   */
  manageInventoryMethodId: number;
  /**
   * The product description.
   * @type {string}
   */
  fullDescription: string;
  /**
   * The product deliveryDateId.
   * @type {number}
   */
  deliveryDateId: number;
  /**
   * The product deliveryDateDisplayOrder.
   * @type {number}
   */
  deliveryDateDisplayOrder: number;
  /**
   * The product deliveryDateName.
   * @type {string}
   */
  deliveryDateName: string;
}
