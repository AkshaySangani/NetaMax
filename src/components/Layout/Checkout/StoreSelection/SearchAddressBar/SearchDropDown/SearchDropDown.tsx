import { ReactElement } from "react";

import { Loader } from "components/common/Loader";
import { AddressCirclePin } from "styled/icons/CheckoutFlow/findStoreWithMap/AddressCirlcePin";
import { ChevronRight } from "styled/icons/CheckoutFlow/findStoreWithMap/ChevronRight";

import { Box, Center, Flex, Text, VStack } from "@chakra-ui/react";

import { ISearchDropDownProps } from "./ISearchDropDownProps";
import { PLACES_STATUS_OK, PLACES_STATUS_ZERO_RESULTS } from "constants/checkoutConstants";

/**
 * The SearchDropDown component.
 * @param { ISearchDropDownProps } props the component props.
 * @returns { ReactElement } The SearchDropDown component.
 */
export const SearchDropDown = (props: ISearchDropDownProps): ReactElement => {
  const { data, status, handleSelectAddress } = props;

  return (
    <VStack width="100%" zIndex="10" backgroundColor="white" paddingX="16px">
      {status === PLACES_STATUS_OK ? (
        data.map((address, i) => (
          <Flex
            onClick={() =>
              handleSelectAddress(address.description, address.structured_formatting.main_text)
            }
            marginTop="0px"
            borderBottom="1px solid rgba(0, 0, 0, .03)"
            paddingY="16px"
            width="100%"
            key={`${address.description}-${i}`}
            justifyContent="space-between"
            alignItems="center"
            cursor="pointer"
          >
            <Flex alignItems="center">
              <AddressCirclePin />
              <Flex marginLeft="8px" direction="column">
                <Flex>
                  <Text fontWeight="600" isTruncated maxWidth="280px">
                    {address.structured_formatting.main_text}
                  </Text>
                </Flex>
                <Text opacity="0.7" isTruncated maxWidth="280px">
                  {address.structured_formatting.secondary_text}
                </Text>
              </Flex>
            </Flex>
            <ChevronRight />
          </Flex>
        ))
      ) : status === PLACES_STATUS_ZERO_RESULTS ? (
        <Box textAlign="center" paddingY="16px">
          <Text fontWeight="600">No se encontraron resultados para la direcci√≥n ingresada</Text>
        </Box>
      ) : (
        <Center height="150px">
          <Loader size="md" delay={300} />
        </Center>
      )}
    </VStack>
  );
};
