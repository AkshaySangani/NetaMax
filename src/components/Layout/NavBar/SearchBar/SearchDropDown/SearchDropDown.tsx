import { ReactElement } from "react";

import { AiOutlineArrowRight } from "react-icons/ai";

import { IncrementalButton } from "components/common/IncrementalButton/IncrementalButton";
import { Loader } from "components/common/Loader";
import { CURRENCY_NAME } from "constants/localeFormatConstants";
import NextLink from "next/link";
import { screenSizes } from "styled/screen";
import { formatMoney } from "utils/currencyUtils";
import { productIsNotAvailable, productMaxQuantity } from "utils/productUtils";

import {
  useMediaQuery,
  Box,
  Button,
  Center,
  Flex,
  Image,
  Link,
  Spacer,
  Text,
} from "@chakra-ui/react";

import { ISearchDropDownProps } from "./ISearchDropDownProps";
import { StyledDropdown } from "./SearchDropDown.styled";

/**
 * The SearchDropDown Component.
 * @param {ISearchDropDownProps} props the component props.
 * @returns {ReactElement} the searchDropDown component.
 */
export const SearchDropDown = (props: ISearchDropDownProps): ReactElement => {
  const {
    value,
    products,
    isLoadingSearchProduct,
    onAddToBasketClick,
    getQtyInBasket,
    onRemoveFromBasketClick,
    clearSearchProducts,
    handleSearchProducts,
    hasMoreProducts,
    emptySearchMessage,
  } = props;

  const [isLargerThanSm] = useMediaQuery(`(min-width: ${screenSizes.smMin}px)`);

  if (!value || (value && value?.length <= 2)) {
    return <></>;
  }
  return isLoadingSearchProduct ? (
    <StyledDropdown $isLargerThanSm={isLargerThanSm}>
      <Center h="100%" w="100%">
        <Loader size="sm" /> Buscando...
      </Center>
    </StyledDropdown>
  ) : (
    <StyledDropdown $isLargerThanSm={isLargerThanSm}>
      {emptySearchMessage !== "" ? (
        <Center h="100%" w="100%">
          <Text>{emptySearchMessage} ðŸ˜”</Text>
        </Center>
      ) : (
        <ul>
          {products &&
            products.map((product) => {
              const qty = getQtyInBasket(product);
              return (
                <li key={product.id}>
                  <Flex width="60%">
                    <NextLink href={`/products/${product.id}`} passHref>
                      <Link
                        onClick={clearSearchProducts}
                        _focus={{
                          boxShadow: "0 0 1px 2px transparent",
                        }}
                        textDecor="none"
                      >
                        <Flex
                          alignItems="center"
                          maxWidth={{ base: qty > 0 ? "200px" : "224px", md: "100%" }}
                        >
                          <Image
                            src={product.seoFilename}
                            height="45px"
                            w="45px"
                            minW={"45px"}
                            fallbackSrc="/assets/images/noimg.png"
                            borderRadius="8px"
                            boxShadow="0px 0px 9px rgb(18 17 17 / 15%)"
                            marginLeft="6px"
                            marginRight="10px"
                            padding="1px"
                          />
                          <Flex flexDir="column">
                            <Text
                              isTruncated
                              fontSize="sm"
                              color="#6A696E"
                              maxWidth={{
                                base: qty > 0 ? "180px" : "224px",
                                md: "300px",
                              }}
                            >
                              {product.name}
                            </Text>
                            <Flex>
                              <Text fontWeight={700}>
                                {formatMoney(parseFloat(product.price), CURRENCY_NAME)}{" "}
                              </Text>
                              {product.oldPrice > product.price && (
                                <Text pl={2} color="gray.500" fontSize="xs" as="del">
                                  {formatMoney(parseFloat(product.oldPrice), CURRENCY_NAME)}
                                </Text>
                              )}
                            </Flex>
                          </Flex>
                        </Flex>
                      </Link>
                    </NextLink>
                    <Box></Box>
                  </Flex>
                  <Spacer />
                  <Flex width={qty > 0 ? "35%" : "15%"} justifyContent="flex-end">
                    {qty > 0 ? (
                      <IncrementalButton
                        min={product.orderMinimumQuantity}
                        max={productMaxQuantity(
                          product.manageInventoryMethodId,
                          product.stockQuantity,
                          product.orderMaximumQuantity
                        )}
                        defaultValue={qty}
                        onAdd={() => onAddToBasketClick(product)}
                        onSubtract={() => onRemoveFromBasketClick(product)}
                      />
                    ) : (
                      <Button
                        data-testid={`add-button-${product.id}`}
                        disabled={productIsNotAvailable(
                          product.manageInventoryMethodId,
                          product.stockQuantity
                        )}
                        onClick={() => {
                          onAddToBasketClick(product);
                        }}
                        height={{ base: "30px", md: "30px" }}
                        width={{ base: "60px", md: "60px" }}
                        _hover={{ bgColor: "none" }}
                        _disabled={{ bgColor: "netaGray.300", cursor: "no-drop" }}
                        colorScheme="netaBlue"
                        borderRadius="full"
                        fontSize="0.8rem"
                      >
                        AÃ±adir
                      </Button>
                    )}
                  </Flex>
                </li>
              );
            })}
          {hasMoreProducts && (
            <li style={{ backgroundColor: "#f3f3f3" }}>
              <Button
                data-testid="hasMoreProductsButton"
                borderRadius="full"
                color="netaBlue.500"
                bg="#E2E2E2"
                onClick={() => {
                  handleSearchProducts();
                }}
              >
                Ir a ver mÃ¡s resultados <AiOutlineArrowRight />
              </Button>
            </li>
          )}
        </ul>
      )}
    </StyledDropdown>
  );
};
