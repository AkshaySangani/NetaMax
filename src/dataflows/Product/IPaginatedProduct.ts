import { IProduct } from "./IProduct";

export interface IPaginatedProduct {
  /**
   * The array of products in the low price offers section.
   * @type {IProduct[]}
   **/
  products: IProduct[];

  /**
   * The items per page low price offers.
   * @type {number}
   **/
  productsPerPage: number;

  /**
   * The pager to low price offers.
   * @type {number}
   **/
  currentPage: number;

  /**
   * The total number of product pages.
   * @type {number}
   */
  totalPages: number;

  /**
   * Is loading the products.
   * @type {boolean}
   **/
  isLoading: boolean;
}
