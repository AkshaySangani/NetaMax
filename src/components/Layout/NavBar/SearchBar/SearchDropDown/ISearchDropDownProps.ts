import { IProduct } from "dataflows/Product/IProduct";

export interface ISearchDropDownProps {
  /**
   * The input value.
   * @type {string}
   */
  value?: string;

  /**
   * The product list.
   * @type {IProduct[]}
   */
  products?: IProduct[];

  /**
   * Indicates if the search products are loading
   * @type {boolean}
   **/
  isLoadingSearchProduct: boolean;

  /**
   * The empty search message
   * @type {string}
   */
  emptySearchMessage: string;

  /**
   * Indicates if the search has more products
   * @type {boolean}
   **/
  hasMoreProducts: boolean;

  /**
   * Clear the search products
   * @type {() => void}
   **/
  clearSearchProducts: () => void;

  /**
   * The add to cart event handler.
   * @type {(product: IProduct) => void}
   */
  onAddToBasketClick: (product: IProduct) => void;

  /**
   * The remove from cart event handler.
   * @type {(product: IProduct) => void}
   */
  onRemoveFromBasketClick: (product: IProduct) => void;

  /**
   * Gets the quantity of the product.
   * @type {(product: IProduct) => number}
   */
  getQtyInBasket: (product: IProduct) => number;

  /**
   * Handles "Ver mÃ¡s" click
   * @type {() => void}
   **/
  handleSearchProducts: () => void;
}
