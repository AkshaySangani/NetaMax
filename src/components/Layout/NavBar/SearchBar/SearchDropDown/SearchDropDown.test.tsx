import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import { IProduct } from "dataflows/Product/IProduct";
import { ISearchDropDownProps } from "./ISearchDropDownProps";
import { SearchDropDown } from "./SearchDropDown";

describe("SearchDropDown Component renders", () => {
  const clearSearchProducts = jest.fn();
  const onAddToBasketClick = jest.fn();
  const onRemoveFromBasketClick = jest.fn();
  const getQtyInBasket = jest.fn((product: IProduct) =>
    ["2426", "2427"].includes(product.id) ? 0 : 10
  );
  const handleSearchProducts = jest.fn();

  let mock: ISearchDropDownProps;

  beforeEach(() => {
    mock = {
      clearSearchProducts,
      onAddToBasketClick,
      onRemoveFromBasketClick,
      getQtyInBasket,
      handleSearchProducts,
      value: "A mocked search value",
      isLoadingSearchProduct: false,
      emptySearchMessage: "",
      hasMoreProducts: false,
      products: [
        {
          id: "2409",
          name: "Medias Noches 8 pz",
          sku: "DSP0010177",
          createdOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          updatedOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          images: [],
          seoFilename: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002737_0.png",
          price: "31.5000",
          oldPrice: "35.0000",
          orderMinimumQuantity: 1,
          orderMaximumQuantity: 15,
          pictureId: 2737,
          categoryId: "3",
          status: "Discount",
          manageInventoryMethodId: 1,
          stockQuantity: 10,
        },
        {
          id: "2425",
          name: "Pan integral Oroweat 12 granos 680 g",
          sku: "DSP0010181",
          createdOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          updatedOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          images: [],
          seoFilename: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002752_0.png",
          price: "47.5000",
          oldPrice: "57.0000",
          orderMinimumQuantity: 1,
          orderMaximumQuantity: 15,
          pictureId: 2752,
          categoryId: "3",
          status: "Discount",
          manageInventoryMethodId: 0,
          stockQuantity: 14,
        },
        {
          id: "2426",
          name: "Pan Integral Multigrano 680 gr",
          sku: "DSP0010182",
          createdOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          updatedOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          images: [],
          seoFilename: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002753_0.png",
          price: "47.5000",
          oldPrice: "57.0000",
          orderMinimumQuantity: 1,
          orderMaximumQuantity: 15,
          pictureId: 2753,
          categoryId: "3",
          status: "Discount",
          manageInventoryMethodId: 1,
          stockQuantity: 0,
        },
        {
          id: "2427",
          name: "1-2-3 Aceite Vegetal 1 Litro",
          sku: "DSP0010189",
          createdOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          updatedOnUtc: new Date("2022-02-22T16:20:47.000Z"),
          images: [],
          seoFilename: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002753_0.png",
          price: "50.5000",
          oldPrice: "57.0000",
          orderMinimumQuantity: 1,
          orderMaximumQuantity: 15,
          pictureId: 2753,
          categoryId: "3",
          status: "Discount",
          manageInventoryMethodId: 0,
          stockQuantity: 0,
        },
      ],
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("SearchDropDown should be render correctly", () => {
    const { container } = render(<SearchDropDown {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("earch more button clicked", () => {
    render(<SearchDropDown {...mock} hasMoreProducts={true} />);
    fireEvent.click(screen.getByTestId("hasMoreProductsButton"));
    expect(mock.handleSearchProducts).toBeCalled();
  });

  test("Search more button should not be in the document", () => {
    render(<SearchDropDown {...mock} />);
    expect(screen.queryByTestId("hasMoreProductsButton")).not.toBeInTheDocument();
  });

  test("Search more button should be in the document", () => {
    render(<SearchDropDown {...mock} hasMoreProducts={true} />);
    expect(screen.getByTestId("hasMoreProductsButton")).toBeInTheDocument();
  });

  test("Add button should be disabled if there is no stock and manageInventoryMethodId is 1", () => {
    render(<SearchDropDown {...mock} />);
    fireEvent.click(screen.getByTestId("add-button-2426"));
    expect(mock.onAddToBasketClick).not.toBeCalled();
  });

  test("Add button should be enabled if there is no stock and manageInventoryMethodId is 0", () => {
    render(<SearchDropDown {...mock} />);
    fireEvent.click(screen.getByTestId("add-button-2427"));
    expect(mock.onAddToBasketClick).toBeCalled();
  });
});

describe("No Result SearchDropDown Component", () => {
  const clearSearchProducts = jest.fn();
  const onAddToBasketClick = jest.fn();
  const onRemoveFromBasketClick = jest.fn();
  const getQtyInBasket = jest.fn(() => 10);
  const handleSearchProducts = jest.fn();

  let mock: ISearchDropDownProps;

  beforeEach(() => {
    mock = {
      clearSearchProducts,
      onAddToBasketClick,
      onRemoveFromBasketClick,
      getQtyInBasket,
      handleSearchProducts,
      value: "This is not an empty search value",
      isLoadingSearchProduct: false,
      emptySearchMessage: "Sorry",
      hasMoreProducts: false,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("SearchDropDown should be render correctly", () => {
    const { container } = render(<SearchDropDown {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Search more button should not be in the document", () => {
    render(<SearchDropDown {...mock} />);
    expect(screen.queryByTestId("hasMoreProductsButton")).not.toBeInTheDocument();
  });
});

describe("Empty SearchDropDown Component", () => {
  const clearSearchProducts = jest.fn();
  const onAddToBasketClick = jest.fn();
  const onRemoveFromBasketClick = jest.fn();
  const getQtyInBasket = jest.fn(() => 10);
  const handleSearchProducts = jest.fn();

  let mock: ISearchDropDownProps;

  beforeEach(() => {
    mock = {
      clearSearchProducts,
      onAddToBasketClick,
      onRemoveFromBasketClick,
      getQtyInBasket,
      handleSearchProducts,
      value: "",
      isLoadingSearchProduct: false,
      emptySearchMessage: "",
      hasMoreProducts: false,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("SearchDropDown should be render correctly", () => {
    const { container } = render(<SearchDropDown {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("SearchDropDown should be render correctly", () => {
    const { container } = render(<SearchDropDown {...mock} value="ab" />);
    expect(container).toMatchSnapshot();
  });

  test("Search more button should not be in the document", () => {
    render(<SearchDropDown {...mock} value="ab" />);
    expect(screen.queryByTestId("hasMoreProductsButton")).not.toBeInTheDocument();
  });
});

describe("Loading SearchDropDown Component", () => {
  const clearSearchProducts = jest.fn();
  const onAddToBasketClick = jest.fn();
  const onRemoveFromBasketClick = jest.fn();
  const getQtyInBasket = jest.fn(() => 10);
  const handleSearchProducts = jest.fn();

  let mock: ISearchDropDownProps;

  beforeEach(() => {
    mock = {
      clearSearchProducts,
      onAddToBasketClick,
      onRemoveFromBasketClick,
      getQtyInBasket,
      handleSearchProducts,
      value: "test value",
      isLoadingSearchProduct: true,
      emptySearchMessage: "",
      hasMoreProducts: false,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("SearchDropDown should be render correctly", () => {
    const { container } = render(<SearchDropDown {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
