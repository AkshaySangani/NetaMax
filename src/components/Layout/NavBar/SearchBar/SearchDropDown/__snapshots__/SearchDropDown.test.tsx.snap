// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty SearchDropDown Component SearchDropDown should be render correctly 1`] = `<div />`;

exports[`Empty SearchDropDown Component SearchDropDown should be render correctly 2`] = `<div />`;

exports[`Loading SearchDropDown Component SearchDropDown should be render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL dZGqEP css-0"
  >
    <div
      class="css-na8a98"
    >
      <div
        class="chakra-spinner css-166ypan"
      >
        <span
          class="css-f8n5zr"
        >
          Loading...
        </span>
      </div>
       Buscando...
    </div>
  </div>
</div>
`;

exports[`No Result SearchDropDown Component SearchDropDown should be render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL dZGqEP css-0"
  >
    <div
      class="css-na8a98"
    >
      <p
        class="chakra-text css-0"
      >
        Sorry
         ðŸ˜”
      </p>
    </div>
  </div>
</div>
`;

exports[`SearchDropDown Component renders SearchDropDown should be render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL dZGqEP css-0"
  >
    <ul>
      <li>
        <div
          class="css-wnav8p"
        >
          <a
            class="chakra-link css-1vgw9ga"
            href="/products/2409"
          >
            <div
              class="css-1897rw9"
            >
              <img
                class="chakra-image__placeholder css-13ixl2"
                src="/assets/images/noimg.png"
              />
              <div
                class="css-j7qwjs"
              >
                <p
                  class="chakra-text css-1lz7kpb"
                >
                  Medias Noches 8 pz
                </p>
                <div
                  class="css-k008qs"
                >
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    $31.50
                     
                  </p>
                  <del
                    class="chakra-text css-z75c2u"
                  >
                    $35.00
                  </del>
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-0"
          />
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-1jknndk"
        >
          <div
            class="chakra-stack css-owwupj"
          >
            <button
              aria-label="remove"
              class="chakra-button css-ljqi8d"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 11h14v2H5z"
                />
              </svg>
            </button>
            <input
              aria-readonly="true"
              class="chakra-input css-xi606m"
              data-testid="counter"
              readonly=""
              value="10"
            />
            <button
              aria-label="add"
              class="chakra-button css-ljqi8d"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 11h-6V5h-2v6H5v2h6v6h2v-6h6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="css-wnav8p"
        >
          <a
            class="chakra-link css-1vgw9ga"
            href="/products/2425"
          >
            <div
              class="css-1897rw9"
            >
              <img
                class="chakra-image__placeholder css-13ixl2"
                src="/assets/images/noimg.png"
              />
              <div
                class="css-j7qwjs"
              >
                <p
                  class="chakra-text css-1lz7kpb"
                >
                  Pan integral Oroweat 12 granos 680 g
                </p>
                <div
                  class="css-k008qs"
                >
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    $47.50
                     
                  </p>
                  <del
                    class="chakra-text css-z75c2u"
                  >
                    $57.00
                  </del>
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-0"
          />
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-1jknndk"
        >
          <div
            class="chakra-stack css-owwupj"
          >
            <button
              aria-label="remove"
              class="chakra-button css-ljqi8d"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 11h14v2H5z"
                />
              </svg>
            </button>
            <input
              aria-readonly="true"
              class="chakra-input css-xi606m"
              data-testid="counter"
              readonly=""
              value="10"
            />
            <button
              aria-label="add"
              class="chakra-button css-ljqi8d"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 11h-6V5h-2v6H5v2h6v6h2v-6h6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="css-wnav8p"
        >
          <a
            class="chakra-link css-1vgw9ga"
            href="/products/2426"
          >
            <div
              class="css-iulr1p"
            >
              <img
                class="chakra-image__placeholder css-13ixl2"
                src="/assets/images/noimg.png"
              />
              <div
                class="css-j7qwjs"
              >
                <p
                  class="chakra-text css-jhf9jw"
                >
                  Pan Integral Multigrano 680 gr
                </p>
                <div
                  class="css-k008qs"
                >
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    $47.50
                     
                  </p>
                  <del
                    class="chakra-text css-z75c2u"
                  >
                    $57.00
                  </del>
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-0"
          />
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-ubxk8c"
        >
          <button
            class="chakra-button css-1rm5qvh"
            data-testid="add-button-2426"
            disabled=""
            type="button"
          >
            AÃ±adir
          </button>
        </div>
      </li>
      <li>
        <div
          class="css-wnav8p"
        >
          <a
            class="chakra-link css-1vgw9ga"
            href="/products/2427"
          >
            <div
              class="css-iulr1p"
            >
              <img
                class="chakra-image__placeholder css-13ixl2"
                src="/assets/images/noimg.png"
              />
              <div
                class="css-j7qwjs"
              >
                <p
                  class="chakra-text css-jhf9jw"
                >
                  1-2-3 Aceite Vegetal 1 Litro
                </p>
                <div
                  class="css-k008qs"
                >
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    $50.50
                     
                  </p>
                  <del
                    class="chakra-text css-z75c2u"
                  >
                    $57.00
                  </del>
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-0"
          />
        </div>
        <div
          class="css-17xejub"
        />
        <div
          class="css-ubxk8c"
        >
          <button
            class="chakra-button css-1rm5qvh"
            data-testid="add-button-2427"
            type="button"
          >
            AÃ±adir
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
